Packs/qualys/Integrations/QualysEventCollector/QualysEventCollector.py

```diff
@@ -626,14 +597,21 @@ def main():  # pragma: no cover
                 first_fetch_time=first_fetch_str,
                 max_fetch=max_fetch_activity_logs,
             )
-            send_events_to_xsiam(activity_logs_events + host_list_detection_events, vendor=VENDOR, product=PRODUCT)
+            send_events_to_xsiam(activity_logs_events, vendor=VENDOR, product=PRODUCT)
 
             # saves next_run for the time fetch-events is invoked
-            last_run.update(logs_next_run)
-            last_run.update(host_next_run)
-            demisto.setLastRun(last_run)
+            demisto.setLastRun(logs_next_run)
+
+        elif command == 'fetch-assets':
+            assets_last_run = demisto.getAssetsLastRun()
+            demisto.debug(f'saved lastrun assets: {assets_last_run}')
+            if skip_fetch_assets(assets_last_run):
+                return
+            demisto.setAssetsLastRun({'assets_last_fetch': time.time()})
+            assets, vulnerabilities = fetch_assets(client=client)
+            send_data_to_xsiam(data=assets, vendor=VENDOR, product='host_detections', data_type='assets')
+            send_data_to_xsiam(data=vulnerabilities, vendor=VENDOR, product='vulnerabilities')
```

Comment:
Instead of saving the last run at the beginning of the run, just sample the time, and only once we are done with EVERYTHING save it to the last run. Imagine that for some reason somewhere in the rest of the execution there is an exception, you will not retry for a whole hour.
```suggestion
            execution_start_time =  time.time()
            assets, vulnerabilities = fetch_assets(client=client)
            send_data_to_xsiam(data=assets, vendor=VENDOR, product='host_detections', data_type='assets')
            send_data_to_xsiam(data=vulnerabilities, vendor=VENDOR, product='vulnerabilities')
            demisto.setAssetsLastRun({'assets_last_fetch': execution_start_time})
```